test/reporters/list.spec.js

```diff
@@ -138,10 +138,16 @@ describe('List reporter', function() {
       var actual = {a: 'actual'};
       var expected = {a: 'expected'};
       var checked = false;
+      var stringifyDiffs = 'stringifyDiffObjs';
       var err;
       test = {};
-      runner.on = runner.once = function(event, callback) {
-        if (!checked && event === 'fail') {
+      runner = createMockRunner('fail', 'fail', null, null, test);
+      runner.on = function(event, callback) {
+        if (
+          callback.toString().includes(stringifyDiffs) && // target correct fail event callback
```

Comment:
What is actual intent here? Are you lazily trying to see if `stringifyDiffObjs` _is_ the callback, or if the callback invokes `stringifyDiffObjs`?

Reply:
Might also migrate this check to the end of the `if` condition. It's more expensive than the other two checks.

Reply:
The latter...callback invokes `stringifyDiffObjs`. Will update var name.